src/Arcus.WebApi.Unit/Security/Authentication/CertificateAuthenticationFilterTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Arcus.WebApi.Security.Authentication;
+using Arcus.WebApi.Unit.Hosting;
+using Arcus.WebApi.Unit.Security.Doubles;
+using Xunit;
+
+namespace Arcus.WebApi.Unit.Security.Authentication
+{
+    public class CertificateAuthenticationFilterTests
+    {
+        private readonly TestApiServer _testServer = new TestApiServer();
+
+        [Fact]
+        public async Task AuthorizedRoute_WithCertificateAuthentication_ShouldFailWithUnauthorized_WhenClientCertificateSubjectNameDoesntMatch()
+        {
+            // Arrange
+            _testServer.AddFilter(new CertificateAuthenticationFilter(X509ValidationRequirement.SubjectName, "subject-name"));
+            _testServer.SetClientCertificate(SelfSignedCertificate.CreateWithSubject("unrecognized-subject-name"));
+
+            using (HttpClient client = _testServer.CreateClient())
+            {
+                var request = new HttpRequestMessage(
+                    HttpMethod.Get,
+                    NoneAuthenticationController.Route);
+
+                // Act
+                using (HttpResponseMessage response = await client.SendAsync(request))
+                {
+                    // Arrange
+                    Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("known-subject", "known-issuername", false)]
+        [InlineData("unrecognizedSubjectName", "known-issuername", true)]
+        [InlineData("known-subject", "unrecognizedIssuerName", true)]
+        [InlineData("unrecognizedSubjectName", "unrecognizedIssuerName", true)]
+        public async Task AuthorizedRoute_WithCertificateAuthentication_ShouldFailWithUnauthorized_WhenAnyClientCertificateValidationDoesntSucceeds(
+            string subjectName,
+            string issuerName,
+            bool expected)
+        {
+            // Arrange
+            _testServer.AddFilter(new CertificateAuthenticationFilter(X509ValidationRequirement.SubjectName, "CN=known-subject"));
```

Comment:
I presume this is similar to how I would configure the global certificate enforcement, right?
Wouldn't it make more sense if I only have to add one of them which takes one or more requirements?

Reply:
From a consumer perspective this would be nicer I think.

Reply:
Ah, I see your point. That could be done and would make the configuration more simpler.

Reply:
You mean to have something like this:

```csharp
new CertificateAuthenticationFilter(new CertificateRequirement(X509ValidationRequirement.SubjectName, "CN=.."),
                                    new CertificateRequirement(X509ValidationRequirement.IssuerName, "CN=kn..."));
```
?

Reply:
Personally I would expect to give one "configuration" object that defines how the validation should work, but maybe a list could be sufficient as well but an object feels more gentle.

Again, personal preference - How we do it is not that important for me, as long as it's nice and simple

Reply:
Yes, I take a look at that.